src/components/ClaimMasterPanel.js

```diff
@@ -50,35 +64,12 @@ class ClaimMasterPanel extends FormPanel {
     );
   }
 
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    if (this._componentDidUpdate(prevProps, prevState, snapshot)) return;
-    if (!prevProps.fetchingClaimCodeCount && this.props.fetchingClaimCodeCount) {
-      this.setState({ claimCodeError: null });
-    } else if (!prevProps.fetchedClaimCodeCount && this.props.fetchedClaimCodeCount) {
-      if (!!this.props.claimCodeCount) {
-        this.setState({ claimCodeError: formatMessage(this.props.intl, "claim", "edit.claimCodeExists") });
-        this.updateAttribute("codeError", true);
-      } else {
-        this.updateAttributes({
-          code: this.state.claimCode,
-          codeError: null,
-        });
-      }
-    }
-  }
-
-  validateClaimCode = (v) => {
-    this.setState(
-      {
-        claimCodeError: null,
-        claimCode: v,
-      },
-      (e) => this.props.validateClaimCode(v),
-    );
+  componentWillUnmount = () => {
+    this.props?.clearClaim();
   };
 
   debounceUpdateCode = _debounce(
```

Comment:
Is this _debounceUpdateCode_ necessary here? _ValidatedTextInput_ has a debounce on his own. Should we remove it?